src/napari_spatialdata/_interactive.py

```diff
@@ -27,6 +28,27 @@ class Interactive:
     None
     """
 
+    def add_element(self, coordinate_system_name: str, element: str) -> SpatialData:
+        for element_type, element_name, _ in self._sdata.filter_by_coordinate_system(
+            coordinate_system_name
+        )._gen_elements():
+            if element_name == element:
+                if element_type == "images":
+                    self._sdata_widget.viewer_model.add_sdata_image(self._sdata, coordinate_system_name, element)
+                elif element_type == "labels":
+                    self._sdata_widget.viewer_model.add_sdata_labels(self._sdata, coordinate_system_name, element)
+                elif element_type == "points":
+                    self._sdata_widget.viewer_model.add_sdata_points(self._sdata, coordinate_system_name, element)
+                elif element_type == "shapes":
+                    if type(self._sdata.shapes[element].iloc[0][0]) == shapely.geometry.point.Point:
+                        self._sdata_widget.viewer_model.add_sdata_circles(self._sdata, coordinate_system_name, element)
+                    elif (type(self._sdata.shapes[element].iloc[0][0]) == shapely.geometry.polygon.Polygon) or (
+                        type(self._sdata.shapes[element].iloc[0][0]) == shapely.geometry.multipolygon.MultiPolygon
+                    ):
+                        self._sdata_widget.viewer_model.add_sdata_shapes(self._sdata, coordinate_system_name, element)
+
```

Comment:
thinking better about this, I think it's not good that we are duplicating the logic of _sdata_widgets.py/_add_shapes(),

I would remove this code and the corresponding code from _add_shapes(); put it in a function; and call that function from here and from _add_shapes(). The function should tell if we are dealing with a point or with a polygon/multipolygon. It could for instance returnt the type directly

Reply:
The check about the first element of the series is more like a temporary hack, and we should replace it with what is being done here: https://github.com/scverse/spatialdata/blob/f58c4f426ab75d070558cf7ec1cb7c03b780d1fa/src/spatialdata/_core/query/_utils.py#L4, so bundling it into a function is more to set a placeholder for the new code, and avoid risking forgetting to update both parts of the code when we apply this change.

Reply:
Actually, let's keep the function as it is now, gonna refactor it in `spatialdata` and then make a PR to this repo.